crowbar_framework/app/models/service_object.rb

```diff
@@ -1271,6 +1271,8 @@ def apply_role(role, inst, in_queue, bootstrap = false)
     pre_cached_nodes = {}
 
     # Each batch is a list of nodes that can be done in parallel.
+    # only run remote chef-clent on a node once
+    batches.uniq!
```

Comment:
> I think flatten should do the trick. Then we can also remove the outside loop.

No. There's a reason to do the run in batches per role (using flatten would break that).  E.g. for the neutron barclamp (and most other barclamps featuring multiple roles)  it's essential that the nodes having the neutron-server role have finished their chef-client run before the run for the neutron-agents nodes is started.